### PR TITLE
Optional arm64 builds [REVPI-2714]

### PR DESCRIPTION
**Pull request by nbuchwitz**
With arm64 support in our kernel and piControl repository, we need to adapt our build scripts as well. This PR introduces optional arm64 support for the kernelbakery. The architecture of choice (eg. `arm` or `arm64`) can be specified with the env var `ARCH`. If the user does not provide `ARCH`, the build script assumes the former default value `arm`

Details about the differences between `arm` and `arm64` builds can be found in the description of the linked ticket.

### CODE REVIEW

**Review by zhan-kunbus — DISMISSED** [review]
LGTM

**Comment by l1k** [comment]
s/gzip/gzip -9/ ?

**Comment by nbuchwitz** [comment]
> s/gzip/gzip -9/ ?

# size comparision
```
-rwxr-xr-x 1 pi pi  7842635 Oct  5 07:50 kernel8.img.gz
-rwxr-xr-x 1 pi pi  7833997 Oct  5 07:57 kernel8.img.gz-9
-rwxr-xr-x 1 pi pi 21277184 Oct  5 07:56 kernel8.img.plain
```
# kernel load time comparision

## CM4S, gzip:

```
006162.477: brfs: File read: /mfs/sd/kernel8.img
006162.513: Loaded 'kernel8.img' to 0x80000 size 0x77ab4b
008767.365: Kernel relocated to 0x200000
008767.391: Device tree loaded to 0x2eff2b00 (size 0xd409)
```

**duration load until relocated: 2604,914 ms**


## CM4S, gzip -9:

```
006326.991: brfs: File read: /mfs/sd/kernel8.img
006327.026: Loaded 'kernel8.img' to 0x80000 size 0x77898d
008926.850: Kernel relocated to 0x200000
008926.876: Device tree loaded to 0x2eff2b00 (size 0xd409)
```

**duration load until relocated: 2599,885 ms**


## CM3, gzip:

```
003710.880: brfs: File read: /mfs/sd/kernel8.img
003710.903: Loaded 'kernel8.img' to 0x80000 size 0x77ab4b
005762.077: Kernel relocated to 0x200000
005762.097: Device tree loaded to 0x2eff7e00 (size 0x81d9)
```

**duration load until relocated: 2051,197 ms**


## CM3, gzip -9:

```
003698.925: brfs: File read: /mfs/sd/kernel8.img
003698.950: Loaded 'kernel8.img' to 0x80000 size 0x77898d
005746.647: Kernel relocated to 0x200000
005746.666: Device tree loaded to 0x2eff7e00 (size 0x81d9)
```

**duration load until relocated: 2047,722 ms**

# TL;DR;

Using `gzip -9` results in a minimal, but measurable improvement in terms of size and boot time. Therefore it can't hurt to use it. Will update the PR.

**Review by l1k — APPROVED** [review]
